examples/EmailerAgent/emailer/agent.py

```diff
@@ -0,0 +1,419 @@
+        allow_frequency_minutes = config.get("allow-frequency-minutes", 60)
+        _allow_frequency_seconds = allow_frequency_minutes * 60
+
+        self.default_config = dict(smtp_address=smtp_address,
+                                   from_address=from_address,
+                                   to_addresses=to_address,
+                                   smtp_port=smtp_port,
+                                   smtp_username=smtp_username,
+                                   smtp_password=smtp_password,
+                                   allow_frequency_minutes=allow_frequency_minutes,
+                                   alert_from_address=from_address,
+                                   alert_to_addresses=to_address,
+                                   send_alerts_enabled=True,
+                                   record_sent_emails=True)
+
+        self.current_config = None
+        self.vip.config.set_default("config", self.default_config)
+
+        self.vip.config.subscribe(self.configure_main,
+                                  actions=["NEW", "UPDATE"], pattern="config")
+
+        # Keep track of keys that have been added to send with.
+        self.tosend = {}
+
+        def onstart(sender, **kwargs):
+            self.vip.pubsub.subscribe('pubsub', topics.PLATFORM_SEND_EMAIL,
+                                      self.on_email_message)
+
+            self.vip.pubsub.subscribe('pubsub', topics.ALERTS_BASE,
+                                      self.on_alert_message)
+            self.vip.pubsub.subscribe('pubsub',
+                                      prefix=topics.ALERTS.format(agent_class='',
+                                                                  agent_uuid=''),
+                                      callback=self.on_alert_message)
+
+        self.core.onstart.connect(onstart, self)
+
+        self.sent_alert_emails = defaultdict(int)
+
+#Add this function to test the emailer by publishing to the topic and passing messages and headers 
+#    @Core.periodic(60)
+#    def accessService(self):
+#        headers = {
+#            "something": 'something'
+#            #"to-addresses": ['alpha.beta@foo.com', 'bob-and-joe@bar.com']
+#        }
+
+#        message = {
+#            "subject": "Something is not right",
+#            "message": "Somethign is down"
+#        }
+
+#        self.vip.pubsub.publish('pubsub', topic='platform/send_email',headers=headers ,message=message)
+
+
+    def _test_smtp_address(self, smtp_address,smtp_port,smtp_username,smtp_password):
+        try:
+            server = smtplib.SMTP(self.current_config.get('smtp_address', None),self.current_config.get('smtp_port', None))
+            server.ehlo()
+            server.starttls()
+            #stmplib docs recommend calling ehlo() before & after starttls()
+            server.ehlo()
+            server.login(self.current_config.get('smtp_username', None), self.current_config.get('smtp_password', None))
```

Comment:
Having this means that we won't be able to use this on campus...as we aren't using a username password for communication.

Please add a use_tls argument to the config and only do the login if we have both smtp_username and smtp_password set.